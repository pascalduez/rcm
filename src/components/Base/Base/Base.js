// @flow

import * as React from 'react';
import { flowRight as compose } from 'lodash-es';

import { withBox, type withBoxProps } from '../Box';
import { withFlex, type withFlexProps } from '../Flex';
import { withText, type withTextProps } from '../Text';

type InjectedProps = withBoxProps & withFlexProps & withTextProps;

export type Props = {
  tagName: string,
  className?: string,
  children?: React.Node,
  refCallback?: (node: React.ElementRef<*>) => {},
} & InjectedProps;

function Base({
  tagName = 'div',
  children,
  refCallback,
  ...extraProps
}: Props) {
  return React.createElement(
    tagName,
    { ref: refCallback, ...extraProps },
    children
  );
}

const Typed = compose(withBox, withFlex, withText)(Base);

Base.displayName = 'Base';
Typed.displayName = 'Base';

export default Typed;
