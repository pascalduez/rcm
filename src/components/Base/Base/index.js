// @flow

export { default as Base } from './Base';
export type { Props as BaseProps } from './Base';
