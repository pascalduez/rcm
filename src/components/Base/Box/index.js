// @flow

export { default as withBox } from './withBox';
export type {
  Props as withBoxProps,
  BackgroundColor as BoxBackgroundColor,
  HoverBackgroundColor as BoxHoverBackgroundColor,
  BorderColor as BoxBorderColor,
  HoverBorderColor as BoxHoverBorderColor,
} from './withBox';
export { default as Box } from './Box';
export type { Props as BoxProps } from './Box';
