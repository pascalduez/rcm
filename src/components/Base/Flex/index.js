// @flow

export { default as withFlex } from './withFlex';
export type { Props as withFlexProps } from './withFlex';
export { default as Flex } from './Flex';
export type { Props as FlexProps } from './Flex';
