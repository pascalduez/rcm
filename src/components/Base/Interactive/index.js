// @flow

export { default as Interactive } from './Interactive';
export type { Props as InteractiveProps } from './Interactive';
