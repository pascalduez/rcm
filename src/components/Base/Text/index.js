// @flow

export { default as withText } from './withText';
export type {
  Props as withTextProps,
  Color as TextColor,
  Theme as TextTheme,
} from './withText';
export { default as Text } from './Text';
export type { Props as TextProps } from './Text';
