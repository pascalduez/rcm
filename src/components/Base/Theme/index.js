// @flow

export { ThemeProvider, ThemeConsumer } from './Theme';
export type { Theme, Props as ThemeProps } from './Theme';
