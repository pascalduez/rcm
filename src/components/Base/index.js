// @flow

export { Text, withText } from './Text';
export type { withTextProps, TextProps, TextColor, TextTheme } from './Text';

export { Box, withBox } from './Box';
export type {
  BoxProps,
  withBoxProps,
  BoxBackgroundColor,
  BoxHoverBackgroundColor,
  BoxBorderColor,
  BoxHoverBorderColor,
} from './Box';

export { Flex, withFlex } from './Flex';
export type { withFlexProps, FlexProps } from './Flex';

export { Base } from './Base';
export type { BaseProps } from './Base';

export { Interactive } from './Interactive';
export type { InteractiveProps } from './Interactive';
