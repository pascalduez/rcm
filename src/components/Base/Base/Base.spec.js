// @flow

import * as React from 'react';
import { render } from 'enzyme';

import Base from './Base';

describe('<Base />', () => {
  it('should render the proper default markup and classes', () => {
    const wrapper = render(<Base />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should pass extra props through', () => {
    const wrapper = render(
      <Base style={{ display: 'block' }}>Just a text</Base>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
