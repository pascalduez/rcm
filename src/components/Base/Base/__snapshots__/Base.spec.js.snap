// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should pass extra props through 1`] = `
<div
  class=""
  style="display:block"
>
  Just a text
</div>
`;

exports[`<Base /> should render the proper default markup and classes 1`] = `
<div
  class=""
/>
`;
